magda-correspondence-api/src/createApiRouter.ts

```diff
@@ -153,16 +154,42 @@ export default function createApiRouter(
             const promise = getDataset(req.params.datasetId).then(dataset => {
                 const dcatDatasetStrings =
                     dataset.aspects["dcat-dataset-strings"];
-
-                const { contactPoint } = dcatDatasetStrings;
+                const datasetPublisher = dataset.aspects["dataset-publisher"];
+                const datasetPublisherEmail =
+                    datasetPublisher &&
+                    datasetPublisher.publisher &&
+                    datasetPublisher.publisher.aspects[
+                        "organization-details"
+                    ] &&
+                    datasetPublisher.publisher.aspects["organization-details"]
+                        .email
+                        ? datasetPublisher.publisher.aspects[
+                              "organization-details"
+                          ].email
+                        : undefined;
+                const emailRegex = /([a-zA-Z0-9._-]+@[a-zA-Z0-9._-]+\.[a-zA-Z0-9._-]+)/;
+                const contactPoint =
+                    (dcatDatasetStrings.contactPoint ||
+                        datasetPublisherEmail) &&
+                    (
+                        dcatDatasetStrings.contactPoint || datasetPublisherEmail
+                    ).match(emailRegex)
+                        ? (
+                              dcatDatasetStrings.contactPoint ||
+                              datasetPublisherEmail
+                          ).match(emailRegex)[1]
+                        : undefined;
 
                 const validContactPoint =
                     contactPoint && emailValidator.validate(contactPoint);
 
                 const recipient = validContactPoint
                     ? contactPoint
                     : options.defaultRecipient;
-
+                if (!validContactPoint) {
+                    body.note =
+                        "You are getting this email because the contact point ‘http:///dept.gov/contact’ on the dataset and ‘phoneus0262222222.’ on the organisation is not a valid email address";
```

Comment:
Was this supposed to have the actual contact point added in? :)

Reply:
Err yes ;) Fixed.